service/src/test/java/uk/gov/hmcts/reform/fpl/service/NoticeOfProceedingsTemplateDataGenerationServiceTest.java

```diff
@@ -0,0 +1,238 @@
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.HIS_HONOUR_JUDGE;
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.MAGISTRATES;
+import static uk.gov.hmcts.reform.fpl.enums.OrderType.CARE_ORDER;
+import static uk.gov.hmcts.reform.fpl.enums.OrderType.EDUCATION_SUPERVISION_ORDER;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createPopulatedApplicants;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createPopulatedChildren;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {NoticeOfProceedingsTemplateDataGenerationService.class})
+@ContextConfiguration(classes = {
+    JacksonAutoConfiguration.class, LookupTestConfig.class, HearingBookingService.class,
+    HearingVenueLookUpService.class, CaseDataExtractionService.class, FixedTimeConfiguration.class
+})
+class NoticeOfProceedingsTemplateDataGenerationServiceTest {
+
+    @Autowired
+    private Time time;
+
+    @Autowired
+    private NoticeOfProceedingsTemplateDataGenerationService noticeOfProceedingsTemplateDataGenerationService;
+
+    private LocalDate futureDate;
+    private List<Element<Child>> children;
+
+    @BeforeEach
+    void setup() {
+        futureDate = time.now().toLocalDate().plusDays(1);
+        children = createPopulatedChildren(time.now().toLocalDate());
+    }
+
+    @Test
+    void shouldApplySentenceFormattingWhenMultipleChildrenExistOnCase() {
+        CaseData caseData = initNoticeOfProceedingCaseData()
+            .children1(children)
+            .orders(Orders.builder()
+                .orderType(ImmutableList.of(CARE_ORDER)).build())
+            .noticeOfProceedings(NoticeOfProceedings.builder()
+                .judgeAndLegalAdvisor(createJudgeAndLegalAdvisor())
+                .proceedingTypes(emptyList())
+                .build())
+            .build();
+
+        DocmosisNoticeOfProceeding templateData = noticeOfProceedingsTemplateDataGenerationService
+            .getTemplateData(caseData);
+        assertThat(templateData.getChildrenNames()).isEqualTo("Bran Stark, Sansa Stark and Jon Snow");
+    }
+
+    @Test
+    void shouldNotApplySentenceFormattingWhenOnlyOneChildExistsOnCase() {
+        CaseData caseData = initNoticeOfProceedingCaseData()
+            .children1(ImmutableList.of(
+                Element.<Child>builder()
+                    .id(UUID.randomUUID())
+                    .value(Child.builder()
+                        .party(ChildParty.builder()
+                            .firstName("Bran")
+                            .lastName("Stark")
+                            .build())
+                        .build())
+                    .build()))
+            .orders(Orders.builder()
+                .orderType(ImmutableList.of(CARE_ORDER)).build())
```

Comment:
minor: same here, we can use `Collections.singletonList` instead of `ImmutableList.of` as its a single value in the list.

Reply:
agree, done.